lib/common/schemas.c

```diff
@@ -9,6 +9,10 @@
 
 #include <crm_internal.h>
 
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE                     // versionsort()
+#endif
```

Comment:
crm_internal.h does this (via portability.h)

Reply:
Didn't know if you wanted to rely on that or be explicit like we do with includes.
```
lib/cib/cib_native.c:14:#  define _GNU_SOURCE
lib/common/scores.c:13:#  define _GNU_SOURCE
lib/common/actions.c:13:#  define _GNU_SOURCE
lib/common/options.c:11:#  define _GNU_SOURCE
lib/common/agents.c:13:#  define _GNU_SOURCE
lib/common/io.c:13:#  define _GNU_SOURCE
lib/common/attrs.c:11:#  define _GNU_SOURCE
lib/common/ipc_attrd.c:11:#  define _GNU_SOURCE
lib/common/procfs.c:13:#  define _GNU_SOURCE
lib/common/ipc_client.c:15:#      define _GNU_SOURCE
lib/common/mainloop.c:13:#  define _GNU_SOURCE
lib/common/pid.c:13:#  define _GNU_SOURCE
lib/common/utils.c:13:#  define _GNU_SOURCE
lib/common/results.c:13:#  define _GNU_SOURCE
lib/common/strings.c:14:#  define _GNU_SOURCE
lib/services/services.c:13:#  define _GNU_SOURCE
lib/services/services_nagios.c:13:#  define _GNU_SOURCE
lib/services/services_lsb.c:13:#  define _GNU_SOURCE
lib/services/services_linux.c:13:#  define _GNU_SOURCE
lib/cluster/membership.c:13:#  define _GNU_SOURCE
```

Reply:
We should rely on crm_internal.h everywhere (though I'm not sure we've ever verified that crm_internal.h is always included everywhere).

Explicit includes are more important in headers than in source files. Eventually I hope to get to one public header and one private header for each library for source files to include.